File: src/commands/ExitCommand.java
package commands;

import client.Application;
import commands.core.ClientSideCommand;
import lib.ExecutionStatus;
import lib.functions.CallbackFunction;

public class ExitCommand extends ClientSideCommand {
    public ExitCommand(CallbackFunction callbackFunction, Application application) {
        super(callbackFunction, application);
    }

    @Override
    protected void executeImplementation(String[] args, CallbackFunction callback) {
        application.stop();
        callback.invoke(ExecutionStatus.SUCCESS, null);
    }
}

File: src/commands/client_side/ExitCommand.kt
package commands.client_side

import application.Application
import commands.client_side.core.ClientSideCommand

class ExitCommand(application: Application) :
    ClientSideCommand(application) {
    override suspend fun executeImplementation(argument: Any?): Result<Unit?> {
        assert(argument == null)

        application.stop()
        return Result.success(null)
    }
}
